_source/_posts/2022-01-30-local-https-java.adoc

```diff
@@ -0,0 +1,278 @@
+
+1. Download the project using the *Generate* button.
+2. Extract the zip file.
+3. Open the project with your favorite IDE.
+
+The last thing to do, is add a basic `RestController` in `src/main/java/com/example/demo/HelloController.java`:
+
+[source,java]
+----
+package com.example.demo;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class HelloController {
+
+    @GetMapping("/")
+    public String sayHello() {
+        return "Yes, this is another Hello World...";
+    }
+}
+----
+
+That's it, start the application! Either from your IDE or by running `./mvnw spring-boot:run`.
+
+Now access this boring, non-secure application in your browser: `\http://localhost:8080/`.
+
+== HTTPS using an Internet proxy
+
+One of the quickest ways to get a local application running with TLS is to use an internet proxy like ngrok.
+https://ngrok.com/download[Download and install ngrok] and then start up the service using:
+
+[source,bash]
+----
+ngrok http 8080
+----
+
+This command will expose your local service running port `8080` over the internet on a _generated_ URL such as `\https://abcd-11-22-33-44.ngrok.io`.
+
+If you don't want to install ngrok, you could use https://localhost.run/[localhost.run], which works using SSH port forwarding; try it out with:
+
+[source,bash]
+----
+ssh -R 80:localhost:8080 nokey@localhost.run
+----
+
+Similar to ngrok, a temporary URL will be generated: `\https://152ccb7ac9369e.lhr.life`
+
+The goal of these tools is to provide a publicly available URL for your local development, which is perfect if you want to share something via the internet. If you don't, it's best not to expose your development machine to any potential attackers online. The following two options avoid this issue.
+
+== Create a certificate authority with mkcert
+
+The next two options use a tool called https://github.com/FiloSottile/mkcert[`mkcert`] which sets up a locally trusted certificate authority (CA), installed into the trust stores on your computer. Any certificates issued by this CA will be trusted by the client of your choice (Chrome, Firefox, curl, etc)
+
+On macOS, you can install `mkcert` with Homebrew; for other operating systems you can find instructions in the https://github.com/FiloSottile/mkcert#installation[`mkcert` readme].
+
+[source,bash]
+----
+brew install mkcert nss
+# nss is only needed if you are using Firefox
+----
+
+Create and install the certificate authority:
+
+====
+[source,bash]
+----
+mkcert -install
+----
+
+[source,txt]
+----
+Created a new local CA 💥
+Sudo password:
+The local CA is now installed in the system trust store! ⚡️
+The local CA is now installed in Java's trust store! ☕️
+----
+====
```

Comment:
This looks fine, but `====` is usually only necessary when you have callouts in the code.